velero-csi-plugin/csi_snapshotter.go

```diff
@@ -125,6 +125,14 @@ func (p *CSISnapshotter) Execute(item runtime.Unstructured, backup *velerov1api.
 	}
 	p.log.Infof("volumesnapshot class=%s", snapshotClass.Name)
 
+	// If deletetionPolicy is not Retain, then in the event of a disaster, the namespace is lost with the volumesnapshot object in it,
+	// the underlying volumesnapshotcontent and the volume snapshot in the storage provider is also deleted.
+	// In such a scenario, the backup objects will be useless as the snapshot handle itself will not be valid.
+	if snapshotClass.DeletionPolicy != snapshotv1beta1api.VolumeSnapshotContentRetain {
+		p.log.Warnf("DeletionPolicy on VolumeSnapshotClass %s is not %s; Deletion of VolumeSnapshot objects will lead to deletion of snapshot in the storage provider; ",
+			snapshotClass.Name, snapshotv1beta1api.VolumeSnapshotContentRetain, snapshotClass.DeletionPolicy, snapshotv1beta1api.VolumeSnapshotContentRetain)
+		p.log.Warnf("Please consider update DeletionPolicy from %s to %s", snapshotClass.DeletionPolicy, snapshotv1beta1api.VolumeSnapshotContentRetain)
```

Comment:
`update` -> `updating`, though the VolumeSnapshotClasses are immutable, so the user basically has to migrate their data to  a new volume.